src/base/bittorrent/session.cpp

```diff
@@ -3816,13 +3810,14 @@ void Session::handleIPFilterError()
     emit IPFilterParsed(true, 0);
 }
 
-void Session::getPendingAlerts(std::vector<lt::alert *> &out, const ulong time)
+std::vector<lt::alert *> Session::getPendingAlerts(const lt::time_duration time) const
 {
-    Q_ASSERT(out.empty());
+    if (time > lt::time_duration::zero())
+        m_nativeSession->wait_for_alert(time);
 
-    if (time > 0)
-        m_nativeSession->wait_for_alert(lt::milliseconds(time));
-    m_nativeSession->pop_alerts(&out);
+    std::vector<lt::alert *> alerts;
+    m_nativeSession->pop_alerts(&alerts);
+    return alerts;
```

Comment:
I hope there is some optimization here that prevents unnecessary copying...

Reply:
https://stackoverflow.com/questions/6233879/move-or-named-return-value-optimization-nrvo

Seems either NRVO and/or move assignment will be involved here so no need worry about it.

Reply:
Sorry my post above wasn't direct enough.
For example: `std::vector<lt::alert *> v = getPendingAlerts();`. In this case `std::vector<lt::alert *> alerts;` in `getPendingAlerts()` will be created and returned. AFAIK `v` will use move constructor to initialize from `alerts`. And thus no copy is involved.